pkg/cloud/services/network/secondarycidr_test.go

```diff
@@ -0,0 +1,264 @@
+		haveSecondaryCIDR bool
+		expect            func(m *mock_ec2iface.MockEC2APIMockRecorder)
+		wantErr           bool
+	}{
+		{
+			name: "Should not associate secondary CIDR if no secondary cidr block info present in control plane",
+		},
+		{
+			name:              "Should return error if unable to describe VPC",
+			haveSecondaryCIDR: true,
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.DescribeVpcs(gomock.AssignableToTypeOf(&ec2.DescribeVpcsInput{})).Return(nil, awserrors.NewFailedDependency("dependency-failure"))
+			},
+			wantErr: true,
+		},
+		{
+			name:              "Should return error if no VPC found",
+			haveSecondaryCIDR: true,
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.DescribeVpcs(gomock.AssignableToTypeOf(&ec2.DescribeVpcsInput{})).Return(nil, nil)
+			},
+			wantErr: true,
+		},
+		{
+			name:              "Should not associate secondary cidr block if already exist in VPC",
+			haveSecondaryCIDR: true,
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.DescribeVpcs(gomock.AssignableToTypeOf(&ec2.DescribeVpcsInput{})).Return(&ec2.DescribeVpcsOutput{
+					Vpcs: []*ec2.Vpc{
+						{
+							CidrBlockAssociationSet: []*ec2.VpcCidrBlockAssociation{
+								{CidrBlock: aws.String("secondary-cidr")},
+							},
+						},
+					}}, nil)
+			},
+		},
+		{
+			name:              "Should associate secondary cidr block if not exist in VPC",
+			haveSecondaryCIDR: true,
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.DescribeVpcs(gomock.AssignableToTypeOf(&ec2.DescribeVpcsInput{})).Return(&ec2.DescribeVpcsOutput{
+					Vpcs: []*ec2.Vpc{
+						{
+							CidrBlockAssociationSet: []*ec2.VpcCidrBlockAssociation{
+								{CidrBlock: aws.String("secondary-cidr-new")},
+							},
+						},
+					}}, nil)
+				m.AssociateVpcCidrBlock(gomock.AssignableToTypeOf(&ec2.AssociateVpcCidrBlockInput{})).Return(nil, nil)
+			},
+		},
+		{
+			name:              "Should return error if failed during associating secondary cidr block",
+			haveSecondaryCIDR: true,
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.DescribeVpcs(gomock.AssignableToTypeOf(&ec2.DescribeVpcsInput{})).Return(&ec2.DescribeVpcsOutput{
+					Vpcs: []*ec2.Vpc{
+						{
+							CidrBlockAssociationSet: []*ec2.VpcCidrBlockAssociation{
+								{CidrBlock: aws.String("secondary-cidr-new")},
+							},
+						},
+					}}, nil)
+				m.AssociateVpcCidrBlock(gomock.AssignableToTypeOf(&ec2.AssociateVpcCidrBlockInput{})).Return(nil, awserrors.NewFailedDependency("dependency-failure"))
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			scheme, err := setupScheme()
+			g.Expect(err).NotTo(HaveOccurred())
+			cl := fake.NewClientBuilder().WithScheme(scheme).Build()
+
+			ec2Mock := mock_ec2iface.NewMockEC2API(mockCtrl)
+
+			mcpScope, err := setupNewManagedControlPlaneScope(cl)
+			g.Expect(err).NotTo(HaveOccurred())
```

Comment:
Can this whole block be refactored to a separate func as used in both the tests?

Reply:
This will couple the two test cases by tying their setup in a function. We have done similar thing across other test cases in CAPA as this helps each test case to evolve separately based on requirement. Wdyt?